docs/CONTROLFLOW.md

```diff
@@ -0,0 +1,319 @@
+# Control Flow Obfuscation
+
+> **Feature is experimental**. To activate it, set environment variable `GARBLE_EXPERIMENTAL_CONTROLFLOW=1`
+
+### Mechanism
+
+
+Control flow obfuscation works in several stages:
+
+1) Collect functions with `//garble:controlflow` comment
+2) Converts [go/ast](https://pkg.go.dev/go/ast) representation to [go/ssa](https://pkg.go.dev/golang.org/x/tools/go/ssa)
+3) Applies [block splitting](#block-splitting)
+4) Generates [junk jumps](#junk-jumps)
+5) Applies [control flow flattening](#control-flow-flattening)
+6) Converts go/ssa back into go/ast
+
+### Example usage
+
+```go
+//garble:controlflow flatten_passes=1 junk_jumps=1 block_splits=1
+func main() {
+	println("Hellow world!")
+}
+```
+
+### Parameter explanation
+
+> Unlike other garble features (which just work), we recommend that you understand how parameters affect control flow obfuscation and what caveats exists.
+
+> Code snippets below without name obfuscation for better readability.
+
+#### Block splitting
+
+Param: `block_splits` (default: `0`)
+
+> Warning: this param affects resulting binary only when used in combination with [flattening](#control-flow-flattening)
+
+
+Block splitting `block_splits` times splits largest block into 2 parts of random size. If there is no suitable block (number of ssa instructions in block is less than 3) work stops without errors.
```

Comment:
Block splitting splits the  largest block into 2 parts of random size, this is done `block_splits` times at most. If there is no more suitable block to split (number of ssa instructions in block is less than 3), splitting stops.